test/jdk/java/lang/StrictMath/InverseTrigTests.java

```diff
@@ -0,0 +1,274 @@
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+import jdk.test.lib.RandomFactory;
+
+/*
+ * @test
+ * @bug 8302026
+ * @key randomness
+ * @library /test/lib
+ * @build jdk.test.lib.RandomFactory
+ * @build Tests
+ * @build FdlibmTranslit
+ * @build InverseTrigTests
+ * @run main InverseTrigTests
+ * @summary Tests for StrictMath.{asin, acos, atan}
+ */
+
+/**
+ * The tests in ../Math/InverseTripTests.java test properties that
+ * should hold for any implementation of the inverse trig functions
+ * ason, acos, and atan, including the FDLIBM-based ones required by
+ * the StrictMath class.  Therefore, the test cases in
+ * ../Math/InverseTrig.java are run against both the Math and
+ * StrictMath versions of the inverse trig methods.  The role of this
+ * test is to verify that the FDLIBM algorithms are being used by
+ * running golden file tests on values that may vary from one
+ * conforming implementation of the hyperbolics to another.
+ */
+
+public class InverseTrigTests {
+    private InverseTrigTests(){}
+
+    public static void main(String... args) {
+        int failures = 0;
+
+        failures += testAgainstTranslitAsin();
+        failures += testAgainstTranslitAcos();
+        failures += testAgainstTranslitAtan();
+
+        if (failures > 0) {
+            System.err.println("Testing the inverse trig functions incurred "
+                               + failures + " failures.");
+            throw new RuntimeException();
+        }
+    }
+
+    // Initialize shared random number generator
+    private static java.util.Random random = RandomFactory.getRandom();
+
+    /**
+     * Test StrictMath.asin against transliteration port of asin.
+     */
+    private static int testAgainstTranslitAsin() {
+        int failures = 0;
+        double x;
+
+        // Test just above subnormal threshold...
+        x = Double.MIN_NORMAL;
+        failures += testRangeAsin(x, Math.ulp(x), 1000);
+
+         // ... and just below subnormal threshold ...
+          x = Math.nextDown(Double.MIN_NORMAL);
+          failures += testRangeAsin(x, -Math.ulp(x), 1000);
+
+         // ... and near 1.0 ...
+          failures += testRangeMidpointAsin(1.0, Math.ulp(x), 2000);
+          // (Note: probes every-other value less than 1.0 due to
+          // change in the size of an ulp at 1.0.
+
+          // Probe near decision points in the FDLIBM algorithm.
+          double[] decisionPoints = {
+               0.5,
+              -0.5,
+
+               0.975,
+              -0.975,
```

Comment:
I think other decision points for asin are 2**-27 and -2**-27 (0x1p-27 and -0x1p-27)